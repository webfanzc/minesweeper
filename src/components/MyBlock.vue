<script setup lang="ts">
import type { BlockState } from '~/types/index'

const blockColors = [
  'text-transparent',
  'text-green-500',
  'text-blue-500',
  'text-yellow-500',
  'text-orange-500',
  'text-red-500',
  'text-purple-500',
  'text-pink-500',
  'text-cyan-500'
]

defineProps<{ block: BlockState }>()

function getBlockClass(block: BlockState) {
  const classes = []

  if (!block.revealed)
    classes.push('bg-gray-100')

  if (block.mine && block.revealed)
    classes.push('border-red bg-red-300/50')
  else if (block.flagged)
    classes.push('text-black')
  else
    classes.push(blockColors[block.adjacentMines] ?? blockColors[blockColors.length - 1])

  return classes.join(' ')
}
</script>
<template>
  <button
    w-10
    h-10
    hover="bg-gray/20"
    border="1 gray-400/20"
    text-4
    leading-none
    :class="getBlockClass(block)"
  >
    {{
      block.revealed || block.flagged ?
        block.flagged ?
          'ðŸš©' :
          block.mine ?
            'ðŸ‘¹' :
            block.adjacentMines
        : ''
    }}
  </button>
</template>
