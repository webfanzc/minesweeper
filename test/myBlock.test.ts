import { mount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'
import MyBlock from '~/components/MyBlock.vue'

describe('MyBlock.vue', () => {
  it('should render', () => {
    const wrapper = mount(MyBlock, {
      props: {
        block: {
          x: 1,
          y: 2,
          flagged: false,
          revealed: false,
          mine: false,
          adjacentMines: 0
        }
      }
    })
    expect(wrapper.text()).toMatchSnapshot('""')
  })

  it('should render mine', () => {
    const wrapper = mount(MyBlock, {
      props: {
        block: {
          x: 1,
          y: 2,
          flagged: false,
          revealed: true,
          mine: true,
          adjacentMines: 0
        }
      }
    })
    expect(wrapper.text()).toMatchInlineSnapshot('"ðŸ‘¹"')
  })

  it('should render flag', () => {
    const wrapper = mount(MyBlock, {
      props: {
        block: {
          x: 1,
          y: 2,
          flagged: true,
          revealed: true,
          mine: false,
          adjacentMines: 0
        }
      }
    })
    expect(wrapper.text()).toMatchSnapshot('"ðŸš©"')
  })

  it('should render adjacentMines', () => {
    const wrapper = mount(MyBlock, {
      props: {
        block: {
          x: 1,
          y: 2,
          flagged: false,
          revealed: true,
          mine: false,
          adjacentMines: 3
        }
      }
    })
    expect(wrapper.text()).toMatchInlineSnapshot('"3"')
  })
})
