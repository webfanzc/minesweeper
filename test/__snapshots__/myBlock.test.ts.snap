// Vitest Snapshot v1

exports[`MyBlock.vue > should render > "" 1`] = `""`;

exports[`MyBlock.vue > should render flag > "🚩" 1`] = `"🚩"`;
